dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/blas/Matrix.java

```diff
@@ -0,0 +1,396 @@
+   */
+  Matrix fill(final float v);
+
+  /**
+   * Reshapes the matrix.
+   * The number of elements must not change.
+   */
+  Matrix reshape(final int newRows, final int newColumns);
+
+  /**
+   * Returns a string representation of this matrix.
+   */
+  String toString();
+
+  /**
+   * Converts the matrix to a one-dimensional array of {@code float}s.
+   */
+  float[] toFloatArray();
+
+  /**
+   * Copies {@code Matrix} {@code m} to this.
+   * @param m a source matrix
+   * @return a source matrix {@code m}
+   */
+  Matrix copy(final Matrix m);
+
+  /**
+   * @return a duplicate of this matrix.
+   */
+  Matrix dup();
+
+  /**
+   * Returns transposed copy of this matrix.
+   */
+  Matrix transpose();
+
+
+  /* Operations */
+
+  /**
+   * Adds a scalar.
+   */
+  Matrix add(final float v);
+
+  /**
+   * Adds a scalar (in place).
+   */
+  Matrix addi(final float v);
+
+  /**
+   * Adds a matrix.
+   */
+  Matrix add(final Matrix m);
+
+  /**
+   * Adds a matrix (in place).
+   */
+  Matrix addi(final Matrix m);
+
+  /**
+   * Adds a vector to all columns of the matrix.
+   */
+  Matrix addColumnVector(final Matrix v);
+
+  /**
+   * Adds a vector to all columns of the matrix (in place).
+   */
+  Matrix addiColumnVector(final Matrix v);
+
+  /**
+   * Adds a vector to all rows of the matrix.
+   */
+  Matrix addRowVector(final Matrix v);
+
+  /**
+   * Adds a vector to all rows of the matrix (in place).
+   */
+  Matrix addiRowVector(final Matrix v);
+
+  /**
+   * Subtracts a scalar.
+   */
+  Matrix sub(final float v);
+
+  /**
+   * Subtracts a scalar (in place).
+   */
+  Matrix subi(final float v);
+
+  /**
+   * Subtracts a matrix.
+   */
+  Matrix sub(final Matrix m);
+
+  /**
+   * Subtracts a matrix (in place).
+   */
+  Matrix subi(final Matrix m);
+
+  /**
+   * Subtracts a vector to all columns of the matrix.
+   */
+  Matrix subColumnVector(final Matrix v);
+
+  /**
+   * Subtracts a vector to all columns of the matrix (in place).
+   */
+  Matrix subiColumnVector(final Matrix v);
+
+  /**
+   * Subtracts a vector to all rows of the matrix.
+   */
+  Matrix subRowVector(final Matrix v);
+
+  /**
+   * Subtracts a vector to all rows of the matrix (in place).
+   */
+  Matrix subiRowVector(final Matrix v);
+
+  /**
+   * (right-)subtracts a scalar.
+   */
+  Matrix rsub(final float v);
+
+  /**
+   * (right-)subtracts a scalar (in place).
+   */
+  Matrix rsubi(final float v);
+
+  /**
+   * (right-)subtracts a matrix.
+   */
+  Matrix rsub(final Matrix m);
+
+  /**
+   * (right-)subtracts a matrix (in place).
+   */
+  Matrix rsubi(final Matrix m);
+
+  /**
+   * Multiplies a scalar.
+   */
+  Matrix mul(final float v);
+
+  /**
+   * Multiplies a scalar (in place).
+   */
+  Matrix muli(final float v);
+
+  /**
+   * Multiplies a matrix.
+   */
+  Matrix mul(final Matrix m);
+
+  /**
+   * Multiplies a matrix (in place).
+   */
+  Matrix muli(final Matrix m);
+
+  /**
+   * Multiplies a vector to all columns of the matrix.
```

Comment:
I'm not sure what this means. Does "multiply a vector to a column" mean calculating the dot product? Please add comments.